packages/nx/src/native/native_hasher.rs

```diff
@@ -97,6 +120,89 @@ fn hash_files(workspace_root: String) -> HashMap<String, String> {
     receiver_thread.join().unwrap()
 }
 
+#[napi]
+fn hash_files_matching_globs(
+    directory: String,
+    glob_patterns: Vec<String>,
+) -> anyhow::Result<Option<String>> {
+    let mut globset_builder = GlobSetBuilder::new();
+
+    for pattern in glob_patterns {
+        globset_builder.add(Glob::new(&pattern).map_err(|_| anyhow!("Invalid Glob {pattern}"))?);
+    }
+    let globset = globset_builder
+        .build()
+        .map_err(|_| anyhow!("Error building globset builder"))?;
+
+    let cpus = available_parallelism().map_or(2, |n| n.get()) - 1;
+
+    let mut walker = WalkBuilder::new(&directory);
+    walker.hidden(false);
+
+    let (sender, receiver) = unbounded::<(String, Vec<u8>)>();
+
+    let receiver_thread = thread::spawn(move || {
+        let mut collection: Vec<FileData> = Vec::new();
+        for (path, content) in receiver {
+            if globset.is_match(&path) {
+                collection.push(FileData {
+                    file: path,
+                    hash: xxh3::xxh3_64(&content).to_string(),
+                });
+            }
+        }
+        collection
+    });
+
+    walker.threads(cpus).build_parallel().run(|| {
+        let tx = sender.clone();
+        let directory = directory.clone();
+        Box::new(move |entry| {
+            use ignore::WalkState::*;
+
+            #[rustfmt::skip]
+                let Ok(dir_entry) = entry else {
+                return Continue;
+            };
+
+            let Ok(content) = std::fs::read(dir_entry.path()) else {
+                return Continue;
+            };
+
+            let Ok(file_path) = dir_entry.path().strip_prefix(&directory) else {
+                return Continue;
+            };
+
+            let Some(file_path) = file_path.to_str() else {
+                return Continue;
+            };
+
+            // convert back-slashes in Windows paths, since the js expects only forward-slash path separators
+            #[cfg(target_os = "windows")]
+            let file_path = file_path.replace('\\', "/");
+
+            tx.send((file_path.to_string(), content)).ok();
+
+            Continue
+        })
+    });
+    drop(sender);
+
+    let mut hashes = receiver_thread.join().unwrap();
+    if hashes.is_empty() {
+        return Ok(None);
+    }
+
+    // Sort the file data so that its in deterministically ordered by file path
+    hashes.sort();
+
+    let sorted_file_hashes: Vec<String> = hashes
+        .iter()
+        .map(|file_data| file_data.hash.clone())
+        .collect();
```

Comment:
We can use `into_iter` here so that we can move ownership from hashes to the sorted_file_hashes. This allows us to remove the clone and remove the need for an additional allocation 🙂 
```suggestion
    let sorted_file_hashes: Vec<String> =
        hashes.into_iter().map(|file_data| file_data.hash).collect();
```